Lib/weakref.py

```diff
@@ -488,6 +488,25 @@ def update(self, dict=None, /, **kwargs):
         if len(kwargs):
             self.update(kwargs)
 
+    def __ior__(self, other):
+        self.update(other)
+        return self
+
+    def __or__(self, other):
+        if isinstance(other, (dict, WeakKeyDictionary, WeakValueDictionary)):
```

Comment:
What's the value of making WeakKeyDict | WeakValueDict return a WeakKeyDict? If that, why not any other Mapping? I can't think of a use case for mixing these like this.

Reply:
So would it make more sense to have `|` accept any mapping and return a WeakKeyDict?

Reply:
Yes.